packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,213 @@
+    IDeputyGuardianModule public deputyGuardianModule;
+
+    /// @notice Used nonces.
+    mapping(bytes32 => bool) public usedNonces;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @param _foundationSafe Address of the Foundation Safe.
+    /// @param _deputyGuardianModule Address of the DeputyGuardianModule used by the SC Safe.
+    /// @param _superchainConfig Address of the SuperchainConfig contract.
+    /// @param _deputy Address of the deputy account.
+    /// @param _deputySignature Signature from the deputy verifying that the account is an EOA.
+    constructor(
+        Safe _foundationSafe,
+        IDeputyGuardianModule _deputyGuardianModule,
+        ISuperchainConfig _superchainConfig,
+        address _deputy,
+        bytes memory _deputySignature
+    )
+        EIP712("DeputyPauseModule", "1")
+    {
+        _setDeputy(_deputy, _deputySignature);
+        deputyGuardianModule = _deputyGuardianModule;
+        FOUNDATION_SAFE = _foundationSafe;
+        SUPERCHAIN_CONFIG = _superchainConfig;
+    }
+
+    /// @notice Getter function for the Foundation Safe address.
+    /// @return foundationSafe_ Foundation Safe address.
+    function foundationSafe() public view returns (Safe foundationSafe_) {
+        foundationSafe_ = FOUNDATION_SAFE;
+    }
+
+    /// @notice Getter function for the SuperchainConfig address.
+    /// @return superchainConfig_ SuperchainConfig address.
+    function superchainConfig() public view returns (ISuperchainConfig superchainConfig_) {
+        superchainConfig_ = SUPERCHAIN_CONFIG;
+    }
+
+    /// @notice Getter function for the Pause message typehash.
+    /// @return pauseMessageTypehash_ Pause message typehash.
+    function pauseMessageTypehash() public pure returns (bytes32 pauseMessageTypehash_) {
+        pauseMessageTypehash_ = PAUSE_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Getter function for the DeputyAuth message typehash.
+    /// @return deputyAuthMessageTypehash_ DeputyAuth message typehash.
+    function deputyAuthMessageTypehash() public pure returns (bytes32 deputyAuthMessageTypehash_) {
+        deputyAuthMessageTypehash_ = DEPUTY_AUTH_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Sets the deputy address.
+    /// @param _deputy Deputy address.
+    /// @param _deputySignature Deputy signature.
+    function setDeputy(address _deputy, bytes memory _deputySignature) external {
+        // Can only be called by the Foundation Safe itself.
+        if (msg.sender != address(FOUNDATION_SAFE)) {
+            revert DeputyPauseModule_NotFromSafe();
+        }
+
+        // Set the deputy address.
+        _setDeputy(_deputy, _deputySignature);
+    }
+
+    /// @notice Sets the DeputyGuardianModule.
+    /// @param _deputyGuardianModule DeputyGuardianModule address.
+    function setDeputyGuardianModule(IDeputyGuardianModule _deputyGuardianModule) external {
```

Comment:
WDYT about enforcing that the new `_deputyGuardianModule` has this contract enabled as a module?

Reply:
The `DeputyGuardianModule` wouldn't have this contract set as a module. The `DeputyGuardianModule` would need to be enabled inside of the Security Council Safe.

Reply:
You're right, wrong phrashing. What I'm getting at is actually enforcing the full auth chain, which would mean verifying the following:

```solidity
Safe(
  DeputyGuardianModule(
    Safe(
      SUPERCHAIN_CONFIG.guardian() // Security Council Safe is the Guardian
    ).getModulesPaginated()[0]     // Returns the DGM address
  ).deputyGuardian()               // Returns the DeputyGuardian (Foundation Ops) Safe
).getModulesPaginated()[0]         // Returns the DPM address
  == address(this)          
```

I think that's a reasonable block of code to drop in, but if not, then I'm fine with this being verified by off-chain tools.

Reply:
My only potential concern here is that if any of these interfaces change then we'll have to modify this contract, which possibly means another re-audit.

Reply:
The only place that would happen is on the DGM, which we control

Reply:
This also assumes that these modules will be the very first module. Mostly just worried about creating dependencies here on external stuff. Same reason I got rid of various checks in the constructor. We can always modify our superchain-ops and deployment scripts but can't easily modify the live contracts.

Reply:
Ok, I can leave it. Still should be verified off chain

Reply:
I like this check. Having it offchain as part of the superchain-ops validations for the task that installs this seems like a good approach